packages/elements-material-ui/src/Popup/index.tsx

```diff
@@ -1,74 +1,72 @@
   t: 'top',
   b: 'bottom',
   l: 'left',
   r: 'right',
   c: 'center',
-  s: 'start',
-  e: 'end',
 };
 
-const preparePosition = (p: any): any => {
-  if (p.charAt(1) === 'c') {
-    return `${positions[p.charAt(0)]}`;
-  } else if (p.charAt(1) === 'l') {
-    return `${positions[p.charAt(0)]}-start`;
-  } else if (p.charAt(0) === 'l' || p.charAt(0) === 'r') {
-    return `${positions[p.charAt(0)]}`;
-  } else {
-    return `${positions[p.charAt(0)]}-end`;
-  }
+//it's for transformOrigin property
+const invertedPositions: any = {
+  t: 'bottom',
+  b: 'top',
+  l: 'right',
+  r: 'left',
+  c: 'center',
 };
 
-export interface ITooltipProps {
-  content: ReactNode;
-  action: 'hover' | 'click' | 'focus';
-  placement: any;
-  trigger: ReactNode;
-}
-
-const mapper = ({ action, position: p, ...rest }: PopupProps): ITooltipProps => {
+const mapper = (props: PopupProps): Omit<PopoverProps, 'open'> => {
+  const { position: p } = props;
+  let aVertical: any = '';
+  let aHorizontal: any = '';
+  let tVertical: any = '';
+  let tHorizontal: any = '';
+  if (p?.charAt(0) === 'l' || p?.charAt(0) === 'r') {
+    aHorizontal = positions[p?.charAt(0)];
+    aVertical = 'center';
+    tHorizontal = invertedPositions[p?.charAt(0)];
+    tVertical = 'center';
+  } else {
+    aVertical = positions[p?.charAt(0) || 'b'];
+    aHorizontal = positions[p?.charAt(1) || 'c'];
+    tVertical = invertedPositions[p?.charAt(0) || 'c'];
+    tHorizontal = invertedPositions[p?.charAt(1) || 'b'];
+  }
   return {
-    ...rest,
-    action: action,
-    placement: p ? preparePosition(p) : 'bottom',
+    anchorOrigin: {
+      vertical: aVertical,
+      horizontal: aHorizontal,
+    },
+    transformOrigin: {
+      vertical: tVertical,
+      horizontal: tHorizontal,
+    },
   };
 };
 
-const Popup = forwardRef<MaterialPopup, PopupProps>((props, ref) => {
-  const mPopupProps = mapper(props);
-  const { action, placement, trigger, content } = mPopupProps;
-  return (
-    <MaterialPopup
-      ref={ref}
-      trigger={action}
-      placement={placement}
-      tooltip={({ tooltipRef, getTooltipProps }) => (
-        <div
-          {...getTooltipProps({
-            ref: tooltipRef,
-            className: 'material-popup-container',
-          })}
-        >
-          {content}
-        </div>
-      )}
-    >
-      {({ getTriggerProps, triggerRef }) =>
-        React.cloneElement(trigger as React.ReactElement<any>, {
-          ...getTriggerProps({
-            ref: triggerRef,
-            className: 'trigger',
-          }),
-        })
-      }
-    </MaterialPopup>
-  );
-});
+export const Popup = forwardRef<HTMLElement, PopupProps>((props, ref) => {
+  const { action, content, trigger } = props;
+  const popupProps: any = mapper(props);
 
-export default Popup;
+  const Component = useMemo(() => {
+    switch (action) {
+      case 'click':
+        return <PopupClick ref={ref} {...popupProps} content={content} trigger={trigger} />;
```

Comment:
Just Return the Component Class. 
useMemo should return PopupClick or PopupHover.
and then you can render the call component with the specific props